build_tools/azure/test_pytest_soft_dependency.sh

```diff
@@ -9,7 +9,7 @@ conda remove -y py pytest || pip uninstall -y py pytest
 if [[ "$COVERAGE" == "true" ]]; then
     # Need to append the coverage to the existing .coverage generated by
     # running the tests
-    CMD="coverage run --append --rcfile=$TEST_DIR/.coveragerc"
+    CMD="coverage run --append --rcfile=.coveragerc"
```

Comment:
I have no idea what is going on but I think that:

```bash
echo -e "[run]\nbranch = True" > .coveragerc
CMD="coverage run --append --rcfile=.coveragerc"
```

should work. I tried to pass the `--branch` flag but it leads to line coverage as well.

Reply:
I'll try ^^

Reply:
BTW feel free to push to my branch if you want to try things!

Reply:
I just activated my azure pipeline. It could be useful for some other PR. Let's see if it works

Reply:
This creates a `.coveragerc` file that only contains `branch=True`. Are we getting the correct coverage information without `source=sklearn`?

Reply:
I think that we can add `source = sklearn` to restrain the coverage. The only thing which need not be there is the `include` flag.

Reply:
Why not use: 

```
CMD="coverage run --append --rcfile=../.coveragerc"
```

This way you do not need to create a new `.coveragerc` file in the `TEST_DIR` folder.

Reply:
Let me give it a try.